docs/contributing-to-airbyte/developing-locally.md

```diff
@@ -0,0 +1,314 @@
+# Developing Locally
+
+Airbyte development is broken into two activities, connector development and platform development. Connector development
+is largely done in Python, though sometimes Java is used for performance reasons. Platform development is done in Java 
+and Kotlin. In addition to the Java and Kotlin code, the Platform also consists of the UI. The UI is developed in 
+TypeScript using React.
+
+## Submitting Code
+
+If you would like to submit code to Airbyte, please follow the [Pull Request Handbook](resources/pull-requests-handbook.md) 
+guide when creating Github Pull Requests. When you are ready to submit code, use the [Submit a New Connector](submit-new-connector.md) document to make 
+sure that the process can go as smoothly as possible.
+
+## Prerequisites
+
+:::info
+
+Manually switching between different language versions can be difficult. We recommend using a version manager such as [`pyenv`](https://github.com/pyenv/pyenv) or [`jenv`](https://github.com/jenv/jenv).
+
+:::
+
+The following technologies are required to build Airbyte locally.
+
+1. [`Java 21`](https://jdk.java.net/archive/)
+2. `Node 20.`
+3. `Python 3.9`
+4. `Docker`
+5. `Jq`
+
+You should also follow the [Quickstart](../using-airbyte/getting-started/oss-quickstart.md) to install `abctl`. In 
+addition to `abctl` you should also install [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installing-from-release-binaries) 
+as well as [Kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+
+If you are looking to build connectors, you should also follow the installation instructions for [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md)
+
+## Connector Contributions
+
+1. [Fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) the [`airbyte`](https://github.com/airbytehq/airbyte) repository.
+2. Clone the fork on your workstation:
+
+```bash
+git clone git@github.com:{YOUR_USERNAME}/airbyte.git
+cd airbyte
+```
+
+3. Make sure that `abctl` is running correctly with the following command `abctl local status`. Verify the status check
+was successful
+
+4. Then, build the connector image:
+    - Verify the `airbyte-ci` tool is installed by running `airbyte-ci --help`, install the command with the instructions in the Prerequisites if the command is not found.
+    - Run `airbyte-ci connectors --name source-<source-name> build` will build your connector image.
+    - Once the command is done, you will find your connector image in your local docker host: `airbyte/source-<source-name>:dev`.
+
+5. You can then load the newly built image into the `abctl` instance using:
+
+```shell
+kind load docker-image airbyte/source-<source-name>:dev -n airbyte-abctl
+```
```

Comment:
So that's how you do it!

@alafanechere - perhaps we could get a new flag to airbyte-ci to do this automatically, e.g. `airbyte-ci connectors --name source-<source-name> --abctl-load build`.  Not a blocking part of this PR

Reply:
I think it's what @perangel is currently working on.

Reply:
We have put support for publishing directly on hold for now. This can be added at some point once we figure out if we are going to bring a registry with us or not.